internal/pullbox/git/git.go

```diff
@@ -20,21 +19,17 @@ func CloneToTmp(repo string) (string, error) {
 	if err := clone(repo, tmpDir); err != nil {
 		return "", errors.WithStack(err)
 	}
-	if err := os.RemoveAll(filepath.Join(tmpDir, ".git")); err != nil {
-		return "", errors.WithStack(err)
-	}
 	return tmpDir, nil
 }
 
 func IsRepoURL(url string) bool {
 	// For now only support ssh
-	return strings.HasPrefix(url, "git@")
+	return strings.HasPrefix(url, "git@") ||
+		(strings.HasPrefix(url, "https://") && strings.HasSuffix(url, ".git"))
 }
 
 func clone(repo, dir string) error {
-	cmd := exec.Command("git", "clone", repo, dir)
-	cmd.Stderr = os.Stderr
-	cmd.Stdout = os.Stdout
+	cmd := ioutil.CommandTTY("git", "clone", repo, dir)
```

Comment:
One motivation to pass in `context.Context` to `devbox.Push` and `devbox.Pull` is so that we set it in this `cmd.Context = ctx`.

Then if the `git` command is taking too long and getting stuck, the user could Ctrl+C and our top-level code would `cancel` the context to also cancel this....

Reply:
That said, I'm not sure what happens (even without explicitly setting a context in this `cmd`):
If the user Ctrl+C terminates the `devbox push/pull` command, then I think that would also terminate this "child process". But I could be wrong.

Reply:
Doing "research", it seems the answer is a bit subtle:
1. if the user interrupts the parent program with SIGINT, then the child processes also terminate (since they share the process group): https://stackoverflow.com/questions/19243431/why-process-created-by-exec-start-quits-if-its-parent-is-killed-by-sigint
2. But this suggests that one needs some manual setup to ensure that all spawened processes are truly killed. Way overkill for our situation: https://stackoverflow.com/questions/71714228/go-exec-commandcontext-is-not-being-terminated-after-context-timeout
3. This way suggests `exec.CommandContext` which is what I had in mind: https://stackoverflow.com/questions/74215347/kill-exec-if-the-parent-process-terminates

Anyway, I'm inclined to say this isn't a problem to be solved right now